libass/ass_directwrite.c

```diff
@@ -577,6 +617,181 @@ static char *get_utf8_name(IDWriteLocalizedStrings *names, int k) {
     return mbName;
 }
 
+static void add_font_face(IDWriteFontFace *face, ASS_FontProvider *provider
+#if DESKTOP_API
```

Comment:
Maybe it's better to use some dummy `ASS_SharedHDC` to reduce such preprocessor logic?

Reply:
I thought about this, but I figured there’s not that much benefit (only this small `#if` and the small `#if` in the `FontPrivate` definition would be removed) and some wasted memory as the drawback. Do you still think that would be better? If so, I’ll change this.

I did unconditionally keep the `IDWriteFont *` field in the same `FontPrivate` struct even though it’s never filled `#if DESKTOP_API` and similarly wastes memory because I figured adding preprocessor logic for _that_ would be more noisy, as the involved (non-preprocessor) logic is more complicated.

Reply:
I think it's better to not have a function with ambiguous number of arguments. I suspect that such one-use static function would be inlined anyway, so there's no runtime cost at all.

Reply:
Oh, you’re suggesting a dummy `SharedHDC` only for this function (but not for `FontPrivate`)? I suppose I can do that.

Reply:
Done: `typedef struct ASS_SharedHDC ASS_SharedHDC` is now done unconditionally, and the function receives `ASS_SharedHDC *` unconditionally, which in the other code path is set to `NULL`.